pkg/handlers/webhookrhobsreceiver.go

```diff
@@ -0,0 +1,252 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/alertmanager/template"
+	log "github.com/sirupsen/logrus"
+
+	oav1alpha1 "github.com/openshift/ocm-agent-operator/api/v1alpha1"
+	"github.com/openshift/ocm-agent/pkg/consts"
+	"github.com/openshift/ocm-agent/pkg/metrics"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	OCMAgentNamespaceName  = "openshift-ocm-agent-operator"
+	AMLabelAlertSourceName = "source"
+	AMLabelAlertSourceHCP  = "HCP"
+	AMLabelAlertSourceDP   = "DP"
+	AMLabelAlertMCID       = "_mc_id"
+	AMLabelAlertHCID       = "_id"
+)
+
+//go:generate mockgen -destination=mocks/webhookrhobsreceiver.go -package=mocks github.com/openshift/ocm-agent/pkg/handlers NewOCMClient
+type NewOCMClient interface {
+	SendServiceLogInFleetMode(n *oav1alpha1.FleetNotification, firing bool, clusterID string) error
+}
+
+type WebhookRHOBSReceiverHandler struct {
+	c   client.Client
+	ocm NewOCMClient
+}
+
+func NewWebhookRHOBSReceiverHandler(c client.Client, o NewOCMClient) *WebhookRHOBSReceiverHandler {
+	return &WebhookRHOBSReceiverHandler{
+		c:   c,
+		ocm: o,
+	}
+}
+
+func (h *WebhookRHOBSReceiverHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// validate request
+	if r != nil && r.Method != http.MethodPost {
+		http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	var err error
+	var alertData AMReceiverData
+	err = json.NewDecoder(r.Body).Decode(&alertData)
+	if err != nil {
+		log.Errorf("Failed to process request body: %s\n", err)
+		http.Error(w, "Bad request body", http.StatusBadRequest)
+		metrics.SetRequestMetricFailure(consts.WebhookReceiverPath)
+		return
+	}
+
+	// process request
+	response := h.processAMReceiver(alertData, r.Context())
+
+	// write response
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(response.Code)
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		log.Errorf("Failed to write to response: %s\n", err)
+		http.Error(w, "Failed to write to response", http.StatusInternalServerError)
+		metrics.SetRequestMetricFailure(consts.WebhookReceiverPath)
+		return
+	}
+
+	metrics.ResetMetric(metrics.MetricRequestFailure)
+}
+
+func (h *WebhookRHOBSReceiverHandler) processAMReceiver(d AMReceiverData, ctx context.Context) *AMReceiverResponse {
+	log.WithField("AMReceiverData", fmt.Sprintf("%+v", d)).Info("Process alert data")
+
+	listOptions := []client.ListOption{
+		client.InNamespace(OCMAgentNamespaceName),
+	}
+	// Let's get all ManagedFleetNotifications
+	mfnl := &oav1alpha1.ManagedFleetNotificationList{}
+	err := h.c.List(ctx, mfnl, listOptions...)
+	if err != nil {
+		log.WithError(err).Error("unable to list managedFleetNotifications")
+		return &AMReceiverResponse{Error: err, Status: "unable to list managedFleetNotifications", Code: http.StatusInternalServerError}
+	}
+
```

Comment:
Now that we're shifting to a design where there's only ever one ManagedFleetNotification per CR, we can simplify this (and the logic in `processAlert`) quite a lot. 

In the non-RHOBS handler it has to search through every `ManagedNotification` looking for the CR that contains the `ManagedNotification` associated with the alert.

In the RHOBS handler we should be able to hopefully do a direct correlation of `the alert defines a managed notification of X, so just get the CR named X` and not do all the list searching - there'll be no need to. That'll hopefully simplify things a bit!